ray-operator/controllers/ray/utils/validation.go

```diff
@@ -37,6 +37,23 @@ func ValidateRayClusterMetadata(metadata metav1.ObjectMeta) error {
 	return nil
 }
 
+func ValidateRayClusterUpgradeOptions(instance *rayv1.RayCluster) error {
+	if instance.Spec.UpgradeStrategy != nil && instance.Spec.UpgradeStrategy.Type != nil &&
+		*instance.Spec.UpgradeStrategy.Type != rayv1.RayClusterRecreate &&
+		*instance.Spec.UpgradeStrategy.Type != rayv1.RayClusterUpgradeNone {
+		return fmt.Errorf("The RayCluster spec is invalid: Spec.UpgradeStrategy.Type value %s is invalid, valid options are %s or %s", *instance.Spec.UpgradeStrategy.Type, rayv1.RayClusterRecreate, rayv1.RayClusterUpgradeNone)
+	}
+
+	// only allow UpgradeStrategy to be set when RayCluster is created directly by user
+	if instance.Spec.UpgradeStrategy != nil && instance.Spec.UpgradeStrategy.Type != nil {
+		creatorCRDType := GetCRDType(instance.Labels[RayOriginatedFromCRDLabelKey])
+		if creatorCRDType == RayJobCRD || creatorCRDType == RayServiceCRD {
+			return fmt.Errorf("upgradeStrategy cannot be set when RayCluster is created by %s", creatorCRDType)
+		}
+	}
+	return nil
+}
+
```

Comment:
Can we refactor this to something like so to make it more concise?


```suggestion
func ValidateRayClusterUpgradeOptions(instance *rayv1.RayCluster) error {
	if instance.Spec.UpgradeStrategy != nil && instance.Spec.UpgradeStrategy.Type != nil {

		if !(*instance.Spec.UpgradeStrategy.Type == rayv1.RayClusterRecreate || *instance.Spec.UpgradeStrategy.Type == rayv1.RayClusterUpgradeNone) {
			return fmt.Errorf("The RayCluster spec is invalid: Spec.UpgradeStrategy.Type value %s is invalid, valid options are %s or %s", *instance.Spec.UpgradeStrategy.Type, rayv1.RayClusterRecreate, rayv1.RayClusterUpgradeNone)
		}

		creatorCRDType := GetCRDType(instance.Labels[RayOriginatedFromCRDLabelKey])
		if creatorCRDType == RayJobCRD || creatorCRDType == RayServiceCRD {
			return fmt.Errorf("upgradeStrategy cannot be set when RayCluster is created by %s", creatorCRDType)
		}
	}
	return nil
}

```